test/lit.cfg

```diff
@@ -958,9 +964,11 @@ if run_vendor == 'apple':
         # Link all "freestanding" tests with -dead_strip, which can effectively
         # even remove parts of the stdlib and runtime, if it's not needed. Since
         # it's a very desired behavior, let's enable it for all executable tests.
-        swift_execution_tests_extra_flags += ' -Xlinker -dead_strip'
+        if not config.swift_freestanding_is_darwin:
+            swift_execution_tests_extra_flags += ' -Xlinker -dead_strip'
 
-        swift_execution_tests_extra_flags += ' -experimental-hermetic-seal-at-link -lto=llvm-full'
+        if not config.swift_freestanding_is_darwin:
```

Comment:
I think this is fine for now, but the hermetic seal at link thing feels like it should have its own feature option sort of thing.

But that being said, I think this is good incremental work. Maybe put in a TODO.